lib/nerves/package/providers/http.ex

```diff
@@ -60,29 +60,32 @@ defmodule Nerves.Package.Providers.HTTP do
     {:error, :nocache}
   end
 
-  defp result({:ok, body}, name, _) do
-    shell_info "Artifact #{name} Downloaded"
-    {:ok, body}
+  defp result({:ok, body}, artifact, _) do
+    shell_info "Artifact #{artifact} Downloaded"
+    file = cache_file(artifact)
+    File.write(file, body)
+    {:ok, file}
   end
   defp result(_, _, []) do
     shell_info "No Available Locations"
     {:error, :nocache}
   end
-  defp result(_, name, locations) do
+  defp result(_, artifact, locations) do
     shell_info "Switching Location"
-    download(name, locations)
+    download(artifact, locations)
   end
 
   defp unpack({:error, _} = error, _, _), do: error
-  defp unpack({:ok, tar}, artifact, destination) do
+  defp unpack({:ok, file}, artifact, destination) do
     shell_info "Unpacking #{artifact}", """
       To Destination:
         #{destination}
     """
     tmp_path = Path.join(destination, ".tmp")
+    File.rm_rf(tmp_path)
     File.mkdir_p!(tmp_path)
     tar_file = Path.join(tmp_path, artifact)
-    File.write(tar_file, tar)
+    File.cp(file, tar_file)
```

Comment:
Can't the `artifact` be untarred as is rather than needing to be copied to a temporary file?